README.md

```diff
@@ -10,3 +11,101 @@ Using `pre-commit` hooks, `flake8`, `black`, `mypy`, `docformatter`, and `pytest
 
 Similarly, Github actions are used to run `flake8`, `black` and `pytest` on every push and pull request. The `pytest` results are sent to [CodeCov](https://about.codecov.io/) using their API for to get test coverage analysis. Details on Github actions are [here](https://github.com/iai-group/guidelines/blob/main/github/Actions.md).
 
+
+# Setting up CI pipeline
+
+This document provides step-by-step instructions for setting up a Continuous Integration (CI) pipeline, including configuring GitHub Actions and integrating a dynamic Gist badge.
+
+---
+
+## There are three CI files in this template repository.
+
+1. `python-package-conda.yaml` this file contains the GitHub action workflow which runs every time someone pushes code to the remote branch.
```

Comment:
Is this really necessary? Assuming that a PR is always created to merge into main, then the CI will run anyway. Note that the contributor to the branch can also run the pre-commit hooks and tests locally.